src/battle_main.c

```diff
@@ -3081,6 +3087,56 @@ void FaintClearSetData(void)
             gBattleMons[i].status2 &= ~(STATUS2_INFATUATED_WITH(gActiveBattler));
         if ((gBattleMons[i].status2 & STATUS2_WRAPPED) && *(gBattleStruct->wrappedBy + i) == gActiveBattler)
             gBattleMons[i].status2 &= ~(STATUS2_WRAPPED);
+
+        // If the fainted mon was holding another Pokemon in Sky Drop, release the mon and clear Sky Drop data
+        if (gActiveBattler == gBattleStruct->skyDropTargets[0] && i == gBattleStruct->skyDropTargets[1])
```

Comment:
based on my other comment about `BattleStruct->skyDropTargets`, you should be able to just get the target to release from `BattleStruct->skyDropTargets[gActiveBattler]`